docs/CODE_GUIDELINES.md

```diff
@@ -341,28 +358,49 @@ int const * const someIntPointer;
 
 ### 5.4. Initialization
 
-Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor to zero-initialize with `{}`.
 
 ✅ Good:
 ```cpp
-int x{};
-int* y{};
-CLog::Log("test: {} {}", x, y);
+int x{3};
+CLog::Log("test: {}", x); // variable used just after its declaration
 ```
 
 ❌ Bad:
 ```cpp
-int x; // used uninitialized
-int* y = nullptr; // default-initialization not using {}
-CLog::Log("test: {} {}", x, y);
+int x{3}; // variable not immediately used, if possible declare the variable close to the point of its use 
+[...many lines of code that do not use variable x...]
+CLog::Log("test: {}", x);
 ```
 
-In general, prefer the `{}` initializer syntax over alternatives. This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
+Although we suggest adopting the use of `{}` for the local variables, like the example above,
+for this use case we allow to initialize the variables also by using the form `=` to improve the code readability.
 
 ✅ Good:
 ```cpp
-int x{5};
-int y{x};
+std::string localVariableX{m_fooVariableOne + m_barVariableTwo}; // less readable
+std::string localVariableX = m_fooVariableOne + m_barVariableTwo; // more readable
+```
+
+In any case, the form `{}` must always be used for members of classes and structures.
+
+When you declare a primitive type variable always initialise it by assigning a value to allow explicit type checking, and to make it even easier to understand the code.
```

Comment:
```suggestion
When you declare a primitive type variable, always initialize it by assigning a value to allow explicit type checking and to make it even easier to understand the code.
```
I personally don't think this rule makes a lot of sense (why now force to use `{nullptr}` over `= nullptr` when above it was given as less readable? why force writing down an obvious value? how would this help type checking?), but if it's the consensus, I won't veto

Reply:
you are talking of a pointer, not a primitive type,
read the first old @garbear comment above